File: DemoDB/Model/Group.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace DemoDB.Model
{
    public class Group
    {
       [Key]
        public int GroupId { get; set; }
        public string GroupName { get; set; }


        public int CreatorId { get; set; }
        public User User { get; set; }

        public DateTime CreatedDate { get; set; }

        [ForeignKey("Group_Id")]
        public ICollection<GroupMember> groupMembers { get; set; }

        [ForeignKey("GroupId")]
        public ICollection<Bill> Bills { get; set; }

        [ForeignKey("GroupId")]
        public ICollection<Transaction> Transactions { get; set; }
    }
}

File: DemoDB/ClientApp/src/app/Model/Group.ts
export class Group {
  constructor(
  ) { }
  public groupId: number;
  public groupName: string;
  public creatorId: number;
  public createdDate;
  public members: number[];
}

export class GroupResponse {
  constructor() { }

  public groupId: number;
  public groupName: string;
  public creatorId: number;
  public creatorName: string;
  public createdDate;
  public members: Grpmember[];
}

export class Grpmember {
  public id: number;
  public name: string;
}
